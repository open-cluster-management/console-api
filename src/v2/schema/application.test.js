/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Application Resolver', () => {
  test('Correctly Resolves Applications Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          application {
            metadata {
              name
              namespace
              creationTimestamp
              labels
              selfLink
            }
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));

  test('Correctly Resolves Single Applications Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          application(name:"gbapp-gbapp",namespace:"default") {
            metadata {
              annotations
              creationTimestamp
              labels
              name
              namespace
              resourceVersion
              selfLink
              uid
            }
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));

  test('Correctly Resolves Application secrets Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          secrets {
            ansibleSecretName
            ansibleSecretNamespace
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
  test('Correctly Resolves Create Application Mutation', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
          createApplication(application:[{
            apiVersion: "mcm.ibm.com/v1alpha1",
            kind: "Application",
            metadata: {
              name: "testapp",
              namespace: "default",
            },
          },
          { deleteLinks: [] },          
        ])
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));

  test('Correctly Resolves Update Application Mutation', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        mutation {
          updateApplication(application:[{
            apiVersion: "mcm.ibm.com/v1alpha1",
            kind: "Application",
            metadata: {
              name: "testapp",
              namespace: "default",
            },
          },
          { deleteLinks: [] },          
        ])
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  }));
});
