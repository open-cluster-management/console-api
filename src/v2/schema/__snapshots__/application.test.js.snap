// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Resolver Correctly Resolves All Applications Query 1`] = `
Object {
  "data": Object {
    "applications": Array [
      Object {
        "metadata": Object {
          "annotations": Object {
            "apps.open-cluster-management.io/applicationrelationships": "gbapp-gbapp-appfrontend,gbapp-gbapp-master,gbapp-gbapp-slave",
            "apps.open-cluster-management.io/dashboard": "https://9.42.81.137:8443/grafana/dashboard/db/gbapp-gbapp-dashboard-via-federated-prometheus",
            "apps.open-cluster-management.io/deployables": "gbapp-gbapp,gbapp-gbapp-redismaster,gbapp-gbapp-redisslave",
            "apps.open-cluster-management.io/placementbindings": "gbapp-gbapp,gbapp-gbapp-redismaster",
            "apps.open-cluster-management.io/subscriptions": "default/gbapp-gbapp-subscription-1",
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2018-11-12T18:48:14Z",
          "labels": Object {
            "app": "gbapp",
            "chart": "gbapp-0.1.0",
            "heritage": "Tiller",
            "name": "gbapp-gbapp",
            "release": "gbapp",
          },
          "name": "gbapp-gbapp",
          "namespace": "default",
          "resourceVersion": "281",
          "selfLink": "/apis/mcm.ibm.com/v1alpha1/namespaces/default/applications/gbapp-gbapp",
          "uid": "8301b18b-e6ab-11e8-b778-9e17a0ceb481",
        },
        "raw": Object {
          "metadata": Object {
            "annotations": Object {
              "apps.open-cluster-management.io/applicationrelationships": "gbapp-gbapp-appfrontend,gbapp-gbapp-master,gbapp-gbapp-slave",
              "apps.open-cluster-management.io/dashboard": "https://9.42.81.137:8443/grafana/dashboard/db/gbapp-gbapp-dashboard-via-federated-prometheus",
              "apps.open-cluster-management.io/deployables": "gbapp-gbapp,gbapp-gbapp-redismaster,gbapp-gbapp-redisslave",
              "apps.open-cluster-management.io/placementbindings": "gbapp-gbapp,gbapp-gbapp-redismaster",
              "apps.open-cluster-management.io/subscriptions": "default/gbapp-gbapp-subscription-1",
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2018-11-12T18:48:14Z",
            "labels": Object {
              "app": "gbapp",
              "chart": "gbapp-0.1.0",
              "heritage": "Tiller",
              "name": "gbapp-gbapp",
              "release": "gbapp",
            },
            "name": "gbapp-gbapp",
            "namespace": "default",
            "resourceVersion": "281",
            "selfLink": "/apis/mcm.ibm.com/v1alpha1/namespaces/default/applications/gbapp-gbapp",
            "uid": "8301b18b-e6ab-11e8-b778-9e17a0ceb481",
          },
          "spec": Object {
            "componentKinds": Array [
              Object {
                "group": "core",
                "kind": "Pods",
              },
            ],
            "descriptor": Object {},
            "selector": Object {
              "matchExpressions": Array [
                Object {
                  "key": "app",
                  "operator": "In",
                  "values": Array [
                    "gbapp",
                    "gbf",
                    "gbrm",
                    "gbrs",
                  ],
                },
              ],
            },
          },
          "status": Object {},
        },
      },
      Object {
        "metadata": Object {
          "annotations": null,
          "creationTimestamp": "2021-04-27T15:05:27Z",
          "labels": null,
          "name": "kevin-helloworld",
          "namespace": "default",
          "resourceVersion": "12070872",
          "selfLink": "/apis/argoproj.io/v1alpha1/namespaces/default/applications/kevin-helloworld",
          "uid": "3ca6512f-0a32-46c2-93cb-a7ad52c2bea3",
        },
        "raw": Object {
          "apiVersion": "argoproj.io/v1alpha1",
          "kind": "Application",
          "metadata": Object {
            "creationTimestamp": "2021-04-27T15:05:27Z",
            "generation": 1586,
            "name": "kevin-helloworld",
            "namespace": "default",
            "resourceVersion": "12070872",
            "selfLink": "/apis/argoproj.io/v1alpha1/namespaces/default/applications/kevin-helloworld",
            "uid": "3ca6512f-0a32-46c2-93cb-a7ad52c2bea3",
          },
          "spec": Object {
            "destination": Object {
              "name": "local-cluster",
              "namespace": "kevin-helloworld-argo",
            },
            "project": "default",
            "source": Object {
              "path": "helloworld",
              "repoURL": "https://github.com/fxiang1/app-samples.git",
              "targetRevision": "HEAD",
            },
            "syncPolicy": Object {
              "automated": Object {
                "prune": true,
                "selfHeal": true,
              },
              "syncOptions": Array [
                "CreateNamespace=true",
              ],
            },
          },
        },
      },
      Object {
        "metadata": Object {
          "annotations": null,
          "creationTimestamp": "2021-04-27T15:08:31Z",
          "labels": null,
          "name": "kevin-helloworld-managed",
          "namespace": "default",
          "resourceVersion": "12072072",
          "selfLink": "/apis/argoproj.io/v1alpha1/namespaces/default/applications/kevin-helloworld-managed",
          "uid": "c41023fa-14cd-4c4a-81b0-7b1719ec33b9",
        },
        "raw": Object {
          "apiVersion": "argoproj.io/v1alpha1",
          "kind": "Application",
          "metadata": Object {
            "creationTimestamp": "2021-04-27T15:08:31Z",
            "generation": 1812,
            "name": "kevin-helloworld-managed",
            "namespace": "default",
            "resourceVersion": "12072072",
            "selfLink": "/apis/argoproj.io/v1alpha1/namespaces/default/applications/kevin-helloworld-managed",
            "uid": "c41023fa-14cd-4c4a-81b0-7b1719ec33b9",
          },
          "spec": Object {
            "destination": Object {
              "name": "ui-remote",
              "namespace": "kevin-helloworld-argo-managed",
            },
            "project": "default",
            "source": Object {
              "path": "helloworld",
              "repoURL": "https://github.com/fxiang1/app-samples.git",
              "targetRevision": "HEAD",
            },
            "syncPolicy": Object {
              "automated": Object {
                "prune": true,
                "selfHeal": true,
              },
              "syncOptions": Array [
                "CreateNamespace=true",
              ],
            },
          },
        },
      },
    ],
  },
}
`;

exports[`Application Resolver Correctly Resolves Application secrets Query 1`] = `
Object {
  "data": Object {
    "secrets": Array [],
  },
}
`;

exports[`Application Resolver Correctly Resolves Applications Query 1`] = `
Object {
  "data": Object {
    "application": Object {
      "metadata": null,
    },
  },
}
`;

exports[`Application Resolver Correctly Resolves Create Application Mutation 1`] = `
Object {
  "data": Object {
    "createApplication": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'statusCode' of undefined",
      "path": Array [
        "createApplication",
      ],
    },
  ],
}
`;

exports[`Application Resolver Correctly Resolves Single Applications Query 1`] = `
Object {
  "data": Object {
    "application": Object {
      "metadata": Object {
        "annotations": Object {
          "apps.open-cluster-management.io/applicationrelationships": "gbapp-gbapp-appfrontend,gbapp-gbapp-master,gbapp-gbapp-slave",
          "apps.open-cluster-management.io/dashboard": "https://9.42.82.240:8443/grafana/dashboard/db/gbapp-gbapp-dashboard-via-federated-prometheus",
          "apps.open-cluster-management.io/deployables": "gbapp-gbapp,gbapp-gbapp-redismaster,gbapp-gbapp-redisslave",
          "apps.open-cluster-management.io/placementbindings": "gbapp-gbapp,gbapp-gbapp-redismaster",
          "apps.open-cluster-management.io/subscriptions": "default/gbapp-gbapp-subscription-1",
          "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
          "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
        },
        "creationTimestamp": "2018-11-12T18:48:14Z",
        "labels": Object {
          "app": "gbapp",
          "chart": "gbapp-0.1.0",
          "heritage": "Tiller",
          "name": "gbapp-gbapp",
          "release": "gbapp",
        },
        "name": "gbapp-gbapp",
        "namespace": "default",
        "resourceVersion": "281",
        "selfLink": "/apis/mcm.ibm.com/v1alpha1/namespaces/default/applications/gbapp-gbapp",
        "uid": "8301b18b-e6ab-11e8-b778-9e17a0ceb481",
      },
    },
  },
}
`;

exports[`Application Resolver Correctly Resolves Update Application Mutation 1`] = `
Object {
  "data": Object {
    "updateApplication": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'statusCode' of undefined",
      "path": Array [
        "updateApplication",
      ],
    },
  ],
}
`;
